tests/helper_test.py

```diff
@@ -76,3 +97,209 @@ def test_megabyte_numbers_are_converted_with_precision_1(self):
     def test_numbers_are_not_converted_to_gigabyte(self):
         readable = helper.humanize_size(1234*1024*1024)
         self.assertEqual(readable, "1234.0M")
+
+
+class TestSplitCommandline(unittest.TestCase):
+
+    def _test(self, base, expected):
+        """Shared helper to reduce some boilerplate."""
+        actual = helper.split_commandline(base)
+        self.assertListEqual(actual, expected)
+
+    def test_simple(self):
+        base = 'echo "foo";sleep 1'
+        expected = ['echo "foo"', 'sleep 1']
+        self._test(base, expected)
+
+    def test_single(self):
+        base = 'echo "foo bar"'
+        expected = [base]
+        self._test(base, expected)
+
+    def test_unicode(self):
+        base = u'echo "foo";sleep 1'
+        expected = ['echo "foo"', 'sleep 1']
+        self._test(base, expected)
+
+
+class TestSplitCommandstring(unittest.TestCase):
+
+    def _test(self, base, expected):
+        """Shared helper to reduce some boilerplate."""
+        actual = helper.split_commandstring(base)
+        self.assertListEqual(actual, expected)
+
+    def test_bytes(self):
+        base = b'echo "foo bar"'
+        expected = [b'echo', b'foo bar']
+        self._test(base, expected)
+
+    def test_unicode(self):
+        base = u'echo "foo €"'
+        expected = [b'echo', u'foo €'.encode('utf-8')]
+        self._test(base, expected)
+
+
+class TestStringSanitize(unittest.TestCase):
+
+    def test_tabs(self):
+        base = 'foo\tbar\noink\n'
+        expected = 'foo' + ' ' * 5 + 'bar\noink\n'
+        actual = helper.string_sanitize(base)
+        self.assertEqual(actual, expected)
+
+
+class TestStringDecode(unittest.TestCase):
+
+    def _test(self, base, expected, encoding='ascii'):
+        actual = helper.string_decode(base, encoding)
+        self.assertEqual(actual, expected)
+
+    def test_ascii_bytes(self):
+        base = u'test'.decode('ascii')
```

Comment:
encode of course. It's not end of frustration that python to implements `str.decode`, which does absolutely nothing. Since a linter can't catch that it in fact does nothing.